src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs

```diff
@@ -55,10 +56,16 @@ public static MauiAppBuilder UseSentry(this MauiAppBuilder builder,
         services.AddSingleton<IConfigureOptions<SentryMauiOptions>, SentryMauiOptionsSetup>();
         services.AddSingleton<Disposer>();
 
-        services.AddSingleton<IMauiElementEventBinder, MauiButtonEventsBinder>();
-        services.AddSingleton<IMauiElementEventBinder, MauiImageButtonEventsBinder>();
-        services.AddSingleton<IMauiElementEventBinder, MauiGestureRecognizerEventsBinder>();
-        services.AddSingleton<IMauiElementEventBinder, MauiVisualElementEventsBinder>();
+        // Resolve the configured options and register any element event binders from these
+        IServiceProvider serviceProvider = services.BuildServiceProvider();
```

Comment:
This will murder startup performance for MAUI apps.  Building the container is one of the most expensive operations in startup.  We're building all of the users services here as well.

Reply:
This will build/assemble all services registered up to that point which will be numerous MAUI services since they get first crack at service collection.  Profiling this won't matter because it depends on the app.  You really only ever want to build the container once

Reply:
Surely not. There's no point in building transient services in advance... even singletons I'd expect it do do lazily (build them as needed). I could see it maybe building a dependency graph in advance... 

In any event, probably need to dig into the internals of ServiceCollection and ServiceProvider to find out.

Reply:
So I checked the source code. Calling `BuildServiceProvider` doesn't, in itself, create any of the services that are registered. It creates an instance of `ServiceProvider`, passing in a bunch of ServiceDescriptors (which come from the the `ServiceCollection`). 

Instances of dependencies get created when you try to resolve them via `IServiceProvider.GetService` and `IServiceProvider.GetRequiredService` etc.

If we're really worried about this, I can use an alternative solution here and/or try to run some benchmarks... from what I can tell, it's not a real problem though.

Reply:
Measuring it will end the debate. 
Even a simple stopwatch just to get an idea of this might help clarify the real impact

Reply:
k - I created some proper benchmarks and the alternative solution to this would be about 33% slower... so merging as is. 

Details in the PR description, so they don't get lost when resolving this discussion.

Reply:
@jamescrosswell @bruno-garcia this is really not good practice, especially in .NET MAUI where we fight very hard for every millisecond of startup performance we can reduce on mobile devices.  Adding more overhead here seems to be unnecessary if it's just about trying to auto instrument another library, where you could instead add another API call to avoid the hit.  Please reconsider this change.

Reply:
https://github.com/getsentry/sentry-dotnet/pull/4208